ddl/column.go

```diff
@@ -379,11 +392,14 @@ func doModifyColumn(t *meta.Meta, job *model.Job, newCol *model.ColumnInfo, oldN
 	return ver, nil
 }
 
-func updateColumn(t *meta.Meta, job *model.Job, newCol *model.ColumnInfo, oldColName *model.CIStr) (ver int64, _ error) {
+func updateColumn(t *meta.Meta, job *model.Job, newCol *model.ColumnInfo, oldColName *model.CIStr, originalTableName string) (ver int64, _ error) {
 	tblInfo, err := getTableInfo(t, job, job.SchemaID)
 	if err != nil {
 		return ver, errors.Trace(err)
 	}
+	if err = checkTableNameChange(t, job, tblInfo.Name.O, originalTableName); err != nil {
```

Comment:
Table name in TiDB is case insensitive, so please compare it by using tblInfo.Name.L